std/net/curl.d

```diff
@@ -1047,9 +1047,8 @@ private auto _basicHTTP(T)(const(char)[] url, const(void)[] sendData, HTTP clien
     };
     client.onReceiveStatusLine = (HTTP.StatusLine l) { statusLine = l; };
     client.perform();
-    enforce!CurlException(statusLine.code / 100 == 2,
-                            format("HTTP request returned status code %d (%s)",
-                                   statusLine.code, statusLine.reason));
+    enforce(statusLine.code / 100 == 2, new HTTPStatusException(statusLine.code,
+            format("HTTP request returned status code %d (%s)", statusLine.code, statusLine.reason)));
```

Comment:
Sth. like below could be handy, but I don't feel strongly about this (I am not using std.net.curl anyways):

```d
@safe pure nothrow
this(int status,
  string file = __FILE__,
  size_t line = __LINE__,
  Throwable next = null)
{
  const msg = format("HTTP request returned status code %d", status);
  super(msg, file, line, next);
  this.status = status;
}

@safe pure nothrow
this(int status,
  string msg,
  string file = __FILE__,
  size_t line = __LINE__,
  Throwable next = null)
{
  const _msg = format("HTTP request returned status code %d (%s)", status, msg);
  super(_msg, file, line, next);
  this.status = status;
}
```